src/App.tsx

```diff
@@ -1,27 +1,81 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import 'bulma/css/bulma.css';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[] | null>(null);
+  const [cathError, setCathError] = useState(false);
+  const handleAllGoodsButton = async () => {
+    try {
+      const goodsFromServer = await getAll();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      setCathError(false);
+      setGoods(goodsFromServer);
+    } catch {
+      setCathError(true);
+    }
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleFirstFiveGoodsButton = async () => {
+    try {
+      const goodsFromServer = await get5First();
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      setCathError(false);
+      setGoods(goodsFromServer);
+    } catch {
+      setCathError(true);
+    }
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const handleRedGoodsButton = async () => {
+    try {
+      const goodsFromServer = await getRedGoods();
+
+      setCathError(false);
+      setGoods(goodsFromServer);
+    } catch {
+      setCathError(true);
+    }
+  };
```

Comment:
for now you have repeated code in 3 functions
will be good to think about the way how you can optimize it